test/integration/scheduler/taint_test.go

```diff
@@ -571,3 +573,359 @@ func TestTaintNodeByCondition(t *testing.T) {
 		})
 	}
 }
+
+func TestTaintBasedEvictionsForNodeUnreachable(t *testing.T) {
+	// Enable TaintNodeByCondition
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.TaintBasedEvictions, true)()
+
+	// Build PodToleration Admission.
+	admission := podtolerationrestriction.NewPodTolerationsPlugin(&pluginapi.Configuration{})
+
+	context := initTestMaster(t, "default", admission)
+
+	// Build clientset and informers for controllers.
+	externalClientset := kubernetes.NewForConfigOrDie(&restclient.Config{
+		QPS:           -1,
+		Host:          context.httpServer.URL,
+		ContentConfig: restclient.ContentConfig{GroupVersion: &schema.GroupVersion{Group: "", Version: "v1"}}})
+	externalInformers := informers.NewSharedInformerFactory(externalClientset, time.Second)
+
+	admission.SetExternalKubeClientSet(externalClientset)
+	admission.SetExternalKubeInformerFactory(externalInformers)
+
+	// Apply feature gates to enable TaintNodesByCondition
+	algorithmprovider.ApplyFeatureGates()
+
+	context = initTestScheduler(t, context, false, nil)
+	cs := context.clientSet
+	informers := context.informerFactory
+	//nsName := context.ns.Name
+
+	// Start NodeLifecycleController for taint.
+	nc, err := nodelifecycle.NewNodeLifecycleController(
+		informers.Coordination().V1beta1().Leases(),
+		informers.Core().V1().Pods(),
+		informers.Core().V1().Nodes(),
+		informers.Apps().V1().DaemonSets(),
+		cs,
+		5*time.Second,    // Node monitor grace period
+		time.Minute,      // Node startup grace period
+		time.Millisecond, // Node monitor period
+		time.Second,      // Pod eviction timeout
+		100,              // Eviction limiter QPS
+		100,              // Secondary eviction limiter QPS
+		50,               // Large cluster threshold
+		0.55,             // Unhealthy zone threshold
+		true,             // Run taint manager
+		true,             // Use taint based evictions
+		false,            // Enabled TaintNodeByCondition feature
+	)
+	if err != nil {
+		t.Errorf("Failed to create node controller: %v", err)
+		return
+	}
+
+	go nc.Run(context.stopCh)
+
+	// Waiting for all controller sync.
+	externalInformers.Start(context.stopCh)
+	externalInformers.WaitForCacheSync(context.stopCh)
+	informers.Start(context.stopCh)
+	informers.WaitForCacheSync(context.stopCh)
+
+	nodeRes := v1.ResourceList{
+		v1.ResourceCPU:    resource.MustParse("4000m"),
+		v1.ResourceMemory: resource.MustParse("16Gi"),
+		v1.ResourcePods:   resource.MustParse("110"),
+	}
+	node1Taints := []v1.Taint{
+		{
+			Key:    schedulerapi.TaintNodeUnreachable,
+			Effect: v1.TaintEffectNoExecute,
+		}}
+
+	node0 := &v1.Node{
```

Comment:
Suggest to use a for loop for nodes initialization. Including creation.

Or, you can reuse `createNodes()`.